frame/nomination-pools/src/lib.rs

```diff
@@ -407,40 +369,67 @@ pub struct PoolMember<T: Config> {
 	/// The quantity of points this member has in the bonded pool or in a sub pool if
 	/// `Self::unbonding_era` is some.
 	pub points: BalanceOf<T>,
-	/// The reward pools total earnings _ever_ the last time this member claimed a payout.
-	/// Assuming no massive burning events, we expect this value to always be below total issuance.
-	/// This value lines up with the [`RewardPool::total_earnings`] after a member claims a
-	/// payout.
-	pub reward_pool_total_earnings: BalanceOf<T>,
+	/// The reward counter at the time of this member's last payout claim.
+	pub last_recorded_reward_counter: T::RewardCounter,
 	/// The eras in which this member is unbonding, mapped from era index to the number of
 	/// points scheduled to unbond in the given era.
 	pub unbonding_eras: BoundedBTreeMap<EraIndex, BalanceOf<T>, T::MaxUnbonding>,
 }
 
 impl<T: Config> PoolMember<T> {
-	fn total_points(&self) -> BalanceOf<T> {
-		self.active_points().saturating_add(self.unbonding_points())
+	/// The pending rewards of this member.
+	fn pending_rewards(
+		&self,
+		current_reward_counter: T::RewardCounter,
+	) -> Result<BalanceOf<T>, Error<T>> {
+		// accuracy note: Reward counters are `FixedU128` with base of 10^18. This value is being
+		// multiplied by a point. The worse case of a point is 10x the granularity of the balance.
+		// Assuming roughly the current issuance of polkadot (12,047,781,394,999,601,455, which is
+		// 1.2 * 10^9 10^10 = 1.2 * 10^19), the worse case point value is around 10^20.
+		//
+		// The final multiplication is:
+		//
+		// rc * 10^20 / 10^18 = rc * 100
+		//
+		// meaning that as long as reward_counter's value is less than 1/100th of its max capacity
+		// (u128), `checked_mul_int` won't saturate.
+		//
+		// given the nature of reward counter being 'pending_rewards / pool_total_point', the only
+		// (unrealistic) way that super high values can be achieved is for a pool to suddenly
+		// receive massive rewards with a very very small amount of stake. In all normal pools, as
+		// the points increase, so does the rewards. Moreover, as long as rewards are not
+		// accumulated for astronomically large durations,
+		// `current_reward_counter.defensive_saturating_sub(self.last_recorded_reward_counter)`
+		// won't be extremely big.
+		(current_reward_counter.defensive_saturating_sub(self.last_recorded_reward_counter))
+			.checked_mul_int(self.active_points())
+			.ok_or(Error::<T>::OverflowRisk)
```

Comment:
Similar to other arithmetic failures, it won't be. If a pool ever gets in a state where this arithmetic is failing, it should be destroyed. 

Currently we don't automatically set the pool to destroying because perhaps this has happened due to a bug.